packages/ui/src/components/DonutViewer.js

```diff
@@ -1,19 +1,77 @@
-import React from 'react'
+import React, { Component } from 'react'
 import PropTypes from 'prop-types'
+import { TransitionGroup } from 'react-transition-group'
 
-const DonutViewer = ({ donuts }) => (
-  <div className='DonutViewer' />
-)
+import Donut from './Donut'
+import Fade from './Fade'
+import '../css/DonutViewer.css'
+
+class DonutViewer extends Component {
+  constructor (...args) {
+    super(...args)
+    this.state = {
+      donuts: []
+    }
+  }
+
+  /**
+   * Handle new donuts via dispatched props.
+   *
+   * @param {Object} newProps
+   * @param {Array[]} newProps.donuts
+   */
+  componentWillReceiveProps ({ donuts }) {
+    const delay = 500
+    const lifespan = 3000
+
+    donuts.forEach(([id, donut], index) => {
+      // Add donuts to state:
+      setTimeout(() => {
+        this.setState({
+          donuts: [...this.state.donuts, [id, donut, index]]
```

Comment:
Adding `index` here to control the `DonutViewer-item-odd` class in `render`. This helps with left-right donut layout on larger screens.